SimplenoteScreenshots/SimplenoteScreenshots.swift

```diff
@@ -41,12 +42,104 @@ class SimplenoteScreenshots: XCTestCase {
+
+        // The index 3 is _intentional_ as that's the desired position of the screenshot in the App
+        // Store
+        snapshot("3-collaborators")
+
+        // Tapping done in the collaborate view dismisses the collaborate screen _and_ the action
+        // menu.
+        doneButton.tap()
+
         let backButton = app.buttons.matching(NSPredicate(format: "label = %@", "Notes")).firstMatch
         XCTAssertTrue(backButton.waitForExistence(timeout: 3))
+        backButton.tap()
+
+        // Before taking a screenshot of the notes, make sure the review prompt header is not on
+        // screen.
+        let reviewPromptButton = app.buttons["I like it"]
+        if reviewPromptButton.waitForExistence(timeout: 3) {
+            reviewPromptButton.tap()
+
+            let dismissReviewButton = app.buttons["No thanks"]
+            XCTAssertTrue(dismissReviewButton.waitForExistence(timeout: 3))
+            dismissReviewButton.tap()
+
+            // also wait for dismiss animation, just in case
+            XCTAssertFalse(dismissReviewButton.waitForExistence(timeout: 3))
+        }
+
+        snapshot("2-all-notes")
+
+        openMenu(using: app)
+
+        let allNotesMenuInput = app.cells.matching(identifier: "all-notes").firstMatch
+        XCTAssertTrue(allNotesMenuInput.waitForExistence(timeout: 3))
+
+        snapshot("4-menu")
+
+        allNotesMenuInput.tap()
+
+        let searchBar = app.otherElements["search-bar"]
+        XCTAssertTrue(searchBar.waitForExistence(timeout: 3))
+        searchBar.tap()
+
+        searchBar.typeText("Recipe")
+
+        let searchCancelButton = app.buttons["Cancel"]
+        XCTAssertTrue(searchCancelButton.waitForExistence(timeout: 3))
 
-        snapshot("0MarkdownPreview")
+        // Dismiss the keyboard before taking the screenshot
+        app.swipeUp()
+        app.swipeDown()
+
+        snapshot("5-search")
+
+        searchCancelButton.tap()
+
+        openMenu(using: app)
+        loadPasscodeScreen(using: app)
+        // Set the passcode
+        // Writing the value here in clear because anyways one can see it being typed.
+        type("1234", onFirstKeyboardOf: app)
+        // Confirm it
+        type("1234", onFirstKeyboardOf: app)
+
+        // Kill the app and relaunch it so we can take a screenshot of the lock screen
+        app.terminate()
+        app.launch()
+
+        // Assuming that if a keyboard is on screen, then the passcode screen has loaded
+        XCTAssertTrue(app.keyboards.firstMatch.waitForExistence(timeout: 10))
+
+        // The screenshot for the passcode should have only 3 characters inserted
+        type("123", onFirstKeyboardOf: app)
+
+        snapshot("6-passcode")
+
+        type("4", onFirstKeyboardOf: app)
+
+        // Now, disable the passcode so we're not blocked by it on next launch.
+        openMenu(using: app)
+        loadPasscodeScreen(using: app)
+        type(ScreenshotsCredentials.testUserPasscode, onFirstKeyboardOf: app)
```

Comment:
This might be overkill in CI, where we reset the Simulator all the time, but I think is worth keeping for the occurrence when devs run this locally. Otherwise, it'd be very annoying to have to type the passcode and then go and disable it after every run.